src/Core/Domain/Shop/Query/SearchShops.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Shop\Query;
+
+use PrestaShop\PrestaShop\Core\Domain\Shop\Exception\ShopException;
+
+/**
+ * Query responsible for getting shops for a given search term
+ */
+class SearchShops
+{
+    /**
+     * @var string
+     */
+    private $searchTerm;
+
+    /**
+     * SearchShops constructor.
+     *
+     * @param string $searchTerm
+     */
+    public function __construct(string $searchTerm)
+    {
+        $this->assertSearchTermNotEmpty($searchTerm);
+
+        $this->searchTerm = $searchTerm;
+    }
+
+    /**
+     * @return string
+     */
+    public function getSearchTerm(): string
+    {
+        return $this->searchTerm;
+    }
+
+    /**
+     * @param string $searchTerm
+     *
+     * @throws ShopException
+     */
+    private function assertSearchTermNotEmpty(string $searchTerm): void
+    {
+        if (empty(trim($searchTerm))) {
+            throw new ShopException('Search term cannot be empty.');
```

Comment:
I think you can create a more precise exception `SearchShopException` that extends `ShopException`